aws/resource_aws_wafv2_web_acl_association_test.go

```diff
@@ -0,0 +1,175 @@
+package aws
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/wafv2"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/terraform"
+)
+
+func TestAccAwsWafv2WebACLAssociation_Basic(t *testing.T) {
+	testName := fmt.Sprintf("web-acl-association-%s", acctest.RandString(5))
+	resourceName := "aws_wafv2_web_acl_association.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafv2WebACLAssociationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsWafv2WebACLAssociationConfig(testName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafv2WebACLAssociationExists(resourceName),
+					testAccMatchResourceAttrRegionalARNNoAccount(resourceName, "resource_arn", "apigateway", regexp.MustCompile(fmt.Sprintf("/restapis/.*/stages/%s", testName))),
+					testAccMatchResourceAttrRegionalARN(resourceName, "web_acl_arn", "wafv2", regexp.MustCompile(fmt.Sprintf("regional/webacl/%s/.*", testName))),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccAWSWafv2WebACLAssociationImportStateIdFunc(resourceName),
+			},
+		},
+	})
+}
+
+func TestAccAwsWafv2WebACLAssociation_Disappears(t *testing.T) {
+	testName := fmt.Sprintf("web-acl-association-%s", acctest.RandString(5))
+	resourceName := "aws_wafv2_web_acl_association.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafv2WebACLAssociationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsWafv2WebACLAssociationConfig(testName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafv2WebACLAssociationExists(resourceName),
+					testAccCheckResourceDisappears(testAccProvider, resourceAwsWafv2WebACLAssociation(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
```

Comment:
we could additionally verify the case where the resource part of the webacl association disappears e.g.
```
{
       Config: testAccAwsWafv2WebACLAssociationConfig(testName),
       Check: resource.ComposeTestCheckFunc(
              testAccCheckAWSWafv2WebACLAssociationExists(resourceName),
	      testAccCheckResourceDisappears(testAccProvider, resourceAwsApiGatewayStage(), "aws_api_gateway_stage.test"),
       ),
       ExpectNonEmptyPlan: true,
},
```